`crucible-concurrency` test suite fails on `master`

Using commit 36cbd02d9eed2b4c76d3d6817c2b9209766a7278, running `cabal test` fails with:

```
$ cabal test crucible-concurrency
Build profile: -w ghc-8.10.4 -O1
In order, the following will be built (use -v for more details):
 - crucible-concurrency-0.1.0.0 (test:crucible-concurrency-golden-tests) (first run)
Preprocessing test suite 'crucible-concurrency-golden-tests' for crucible-concurrency-0.1.0.0..
Building test suite 'crucible-concurrency-golden-tests' for crucible-concurrency-0.1.0.0..
[1 of 1] Compiling Main             ( test/Main.hs, /home/rscott/Documents/Hacking/Haskell/sandbox/crucible/dist-newstyle/build/x86_64-linux/ghc-8.10.4/crucible-concurrency-0.1.0.0/t/crucible-concurrency-golden-tests/build/crucible-concurrency-golden-tests/crucible-concurrency-golden-tests-tmp/Main.o )

test/Main.hs:69:69: error:
    • Couldn't match expected type ‘msgs0 -> Crux.SayWhat’
                  with actual type ‘Maybe Crux.CruxOptions’
    • Possible cause: ‘Just’ is applied to too many arguments
      In the second argument of ‘($)’, namely ‘Just (fst options)’
      In the expression:
        Crux.mkOutputConfig False outh outh $ Just (fst options)
      In the expression:
        do let options = ...
           let ?outputConfig = Crux.mkOutputConfig False outh outh
                                 $ Just (fst options)
           run options
   |
69 |           let ?outputConfig = Crux.mkOutputConfig False outh outh $ Just (fst options)
   |                                                                     ^^^^^^^^^^^^^^^^^^

test/Main.hs:70:11: error:
    • Couldn't match type ‘Maybe Crux.CruxOptions
                           -> Crux.OutputConfig msgs0’
                     with ‘Crux.OutputConfig Crux.CruxLogMessage’
        arising from a functional dependency between constraints:
          ‘?outputConfig::Crux.OutputConfig Crux.CruxLogMessage’
            arising from a use of ‘run’ at test/Main.hs:70:11-21
          ‘?outputConfig::Maybe Crux.CruxOptions -> Crux.OutputConfig msgs0’
            arising from the implicit-parameter binding for ?outputConfig at test/Main.hs:(61,8)-(70,21)
    • In a stmt of a 'do' block: run options
      In the expression:
        do let options = ...
           let ?outputConfig = Crux.mkOutputConfig False outh outh
                                 $ Just (fst options)
           run options
      In the second argument of ‘($)’, namely
        ‘\ outh
           -> do let ...
                 let ...
                 ....’
   |
70 |           run options
   |           ^^^^^^^^^^^
```

This is likely due to #762.

--- a/crucible-concurrency/.gitignore
+++ b/crucible-concurrency/.gitignore
@@ -1,0 +1,5 @@
+examples/cbl/*.out
+sv-benchmarks/c/pthread-divine/*.out
+sv-benchmarks/c/pthread-ext/*.out
+sv-benchmarks/c/pthread-lit/*.out
+sv-benchmarks/c/pthread/*.out

--- a/crucible-concurrency/test/Main.hs
+++ b/crucible-concurrency/test/Main.hs
@@ -66,5 +66,6 @@ testSimulator inFile outFile =
                                               Crux.checkPathSat = True
                                             },
                          defaultCrucesOptions)
-          let ?outputConfig = Crux.mkOutputConfig False outh outh $ Just (fst options)
+          let ?outputConfig = Crux.mkOutputConfig False outh outh
+                                Crux.cruxLogMessageToSayWhat $ Just (fst options)
           run options

